extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-sra.js

```diff
@@ -0,0 +1,471 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {AmpAd3PImpl} from '../../../amp-ad/0.1/amp-ad-3p-impl';
+import {
+  AmpA4A,
+  RENDERING_TYPE_HEADER,
+  XORIGIN_MODE,
+} from '../../../amp-a4a/0.1/amp-a4a';
+import {Services} from '../../../../src/services';
+import {
+  AmpAdNetworkDoubleclickImpl,
+  getNetworkId,
+  constructSRABlockParameters,
+  TFCD,
+  resetSraStateForTesting,
+} from '../amp-ad-network-doubleclick-impl';
+import {
+  MANUAL_EXPERIMENT_ID,
+} from '../../../../ads/google/a4a/traffic-experiments';
+import {
+  EXPERIMENT_ATTRIBUTE,
+} from '../../../../ads/google/a4a/utils';
+import {utf8Encode} from '../../../../src/utils/bytes';
+import {BaseElement} from '../../../../src/base-element';
+import {createElementWithAttributes} from '../../../../src/dom';
+import {layoutRectLtwh} from '../../../../src/layout-rect';
+import {Xhr, FetchResponseHeaders} from '../../../../src/service/xhr-impl';
+import {dev} from '../../../../src/log';
+import '../../../amp-ad/0.1/amp-ad';
+import * as sinon from 'sinon';
+
+function setupForAdTesting(win) {
+  const doc = win.document;
+  // TODO(a4a-cam@): This is necessary in the short term, until A4A is
+  // smarter about host document styling.  The issue is that it needs to
+  // inherit the AMP runtime style element in order for shadow DOM-enclosed
+  // elements to behave properly.  So we have to set up a minimal one here.
+  const ampStyle = doc.createElement('style');
+  ampStyle.setAttribute('amp-runtime', 'scratch-fortesting');
+  doc.head.appendChild(ampStyle);
+}
+
+describes.realWin('amp-ad-network-doubleclick-impl', {amp: true,
+  allowExternalResources: true}, env => {
```

Comment:
Add comment about why you need allowExternalResources.  We should also consider creating a base A4A experiment class in order to isolate/share testing framework setup across A4A tests.